src/components/sortable_table/index.js

```diff
@@ -0,0 +1,105 @@
+var React = window.React || require('react/addons');
+var _ = require('lodash');
+var TableHeader = require('./header');
+var TableRow = require('./row');
+var SortableStyles = require('../../styles/sortable_table');
+
+/*
+ * Takes an array of json objects (for example, a Backbone.Collection.toJSON())
+ * and an array of attributes to serve as column header labels and creates a sortable table from the data.
+ * Column header labels are clickable and will trigger sort either in
+ * ascending or descending order based on which option is currently active.
+ *
+ * Responsible for managing the expanded/condensed state of rows and for proxying sort
+ * events up to parent view for processing.
+ */
+
+var SortableTable = React.createClass({
+
+  propTypes: {
+    tableType: React.PropTypes.string,
+    label: React.PropTypes.string,
+    collection: React.PropTypes.arrayOf(React.PropTypes.object).isRequired,
+    columnNames: React.PropTypes.arrayOf(React.PropTypes.string).isRequired,
+    onSortCollection: React.PropTypes.func.isRequired,
+    isBulkEditable: React.PropTypes.bool,
+    onBulkSelect: React.PropTypes.func,
+    modelChangerUtilities: React.PropTypes.object,
+    navigatorUtility: React.PropTypes.object
+  },
+
+  mixins: [SortableStyles],
+
+  getDefaultProps: function() {
+    return {
+      isBulkEditable: false,
+      onBulkSelect: _.noop(),
+      modelChangerUtilities: {}
+    };
+  },
+
+  getInitialState: function() {
+    return {
+      expanded: 'condensed',
+      sortBy: 'number'
+    };
+  },
+
+  onExpandClicked: function(expandOrCondense) {
+    this.setState({
+      expanded: expandOrCondense
+    });
+  },
+
+  render: function() {
+    var tableName = this.props.label.charAt(0).toUpperCase() + this.props.label.slice(1);
+    var tableClass = "sortable-table " + this.props.label;
+    var rows = [];
+
+    _.each(this.props.collection, function(model) {
+      var modelId = [model.product.id, model.number];
+      if (model.isMatched && !model.parent) {
+        // Add a spacer row above matched parents.
+        rows.push(
+          <tr key={modelId+':spacer'} className="spacer" style={SortableStyles.row.spacer} />
+        );
+      }
+      rows.push(
+        <TableRow
+          key={modelId}
+          model={model}
+          columns={this.props.columnNames}
+          expanded={this.state.expanded}
+          modelChangerUtilities={this.props.modelChangerUtilities}
+          navigatorUtility={this.props.navigatorUtility}
+          isBulkEditable={this.props.isBulkEditable}
+          onBulkSelect={this.props.onBulkSelect}
```

Comment:
For big property lists like this (and the one below in `<TableHeader/>`,) I find that it's better for readability to use an object and splat assignment:

```
var props = {
  key: modelId,
  model: model,
  columns: this.props.columnNames,
  expanded: this.state.expanded,
  modelChangerUtilities: this.props.modelChangerUtilities,
  navigatorUtility: this.props.navigatorUtility,
  isBulkEditable: this.props.isBulkEditable,
  onBulkSelect: this.props.onBulkSelect
};

<TableRow {...props} />
```

NB this is not a "must change" comment by any means, more of a "food for thought" :apple: :bread: